Source/monstdat.h

```diff
@@ -61,17 +61,299 @@ enum class MonsterClass : uint8_t {
 	Animal,
 };
 
-enum monster_resistance : uint8_t {
-	// clang-format off
-	RESIST_MAGIC     = 1 << 0,
-	RESIST_FIRE      = 1 << 1,
-	RESIST_LIGHTNING = 1 << 2,
-	IMMUNE_MAGIC     = 1 << 3,
-	IMMUNE_FIRE      = 1 << 4,
-	IMMUNE_LIGHTNING = 1 << 5,
-	IMMUNE_NULL_40   = 1 << 6,
-	IMMUNE_ACID      = 1 << 7,
-	// clang-format on
+struct MonsterResists {
+	// NOTE: unless we remove backwards compatability, or maybe use up some extra memory locations,
+	// magic, fire, and lightning resistances should be less than 128, otherwise they will not save properly.
+	uint8_t Magic = 0;
+	uint8_t Fire = 0;
+	uint8_t Lightning = 0;
+	uint8_t Acid = 0;
+
+	uint8_t getMagicResist()
+	{
+		return this->Magic;
+	}
+	uint8_t getFireResist()
+	{
+		return this->Fire;
+	}
+	uint8_t getLightningResist()
+	{
+		return this->Lightning;
+	}
+	uint8_t getAcidResist()
+	{
+		return this->Acid;
+	}
+
+	/**
+	 * @brief Returns if monster has any resistances (or immunities) to Magic, Fire, or Lightning.
+	 * @return True if any of the monster's non-acid resistances are greater than 0.
+	 */
+	bool hasResistancesOrImmunities() const
+	{
+		return this->Magic > 0 || this->Fire > 0 || this->Lightning > 0;
+	}
+	/**
+	 * @brief Returns if monster has any resistances to Magic, Fire, or Lightning. Must have at least one greater than 0 but less than 100.
+	 * @return True if any of the monster's non-acid resistances are greater than 0 but less than 100.
+	 */
+	bool hasResistances() const
+	{
+		return (this->Magic > 0 && this->Magic < 100) || (this->Fire > 0 && this->Fire < 100) || (this->Lightning > 0 && this->Lightning < 100);
+	}
+
+	/**
+	 * @brief Returns if monster has any immunities to Magic, Fire, or Lightning.
+	 * @return True if monster has non-acid immunities.
+	 */
+	bool hasImmunities() const
+	{
+		return this->Magic >= 100 || this->Fire >= 100 || this->Lightning >= 100;
+	}

```

Comment:
This seams specific (ignores a specific element type) to the UI and probably shouldn't be part of the general class.